os/board/rtl8721csm/src/rtl8721csm_boot.c

```diff
@@ -85,6 +90,45 @@ extern FAR struct gpio_lowerhalf_s *amebad_gpio_lowerhalf(u32 pinname, u32 pinmo
 /************************************************************************************
  * Public Functions
  ************************************************************************************/
+void board_i2s_initialize(void)
+{
+#if defined(CONFIG_AUDIO_I2SCHAR) && defined(CONFIG_AMEBAD_I2S)
+	i2schar_devinit();
+#endif
+}
+
+void board_spi_initialize(void)
+{
+#ifdef CONFIG_SPI
+	struct spi_dev_s *spi;
+	spi = up_spiinitialize(1);
+
+#ifdef CONFIG_SPI_USERIO
+	if (spi_uioregister(1, spi) < 0) {
+		lldbg("Failed to register SPI%d\n", 1);
+	}
+#endif
+#endif
+}
+
+void board_i2c_initialize(void)
+{
+#ifdef CONFIG_I2C
+	FAR struct i2c_dev_s *i2c;
+	int bus = 0;
+	int ret;
+	char path[16];
+
+	snprintf(path, sizeof(path), "/dev/i2c-%d", bus);
+	i2c = up_i2cinitialize(bus);
+	if (i2c) {
+		ret = i2c_uioregister(path, i2c);
+		if (ret < 0) {
+			up_i2cuninitialize(i2c);
+		}
+	}
```

Comment:
CONFIG_I2C_USERIO should be here.

Reply:
could you please check same kind of uio in spi and i2s??

Reply:
CONFIG_SPI_USERIO was checked in the commit, I didn't find any reference to CONFIG_I2S_USERIO. Do I still need to check the flag anyway?